syntax = "proto3";

import "proto/events.proto";

message ProjectData {
    string name = 1;
    string id = 2;
    int32 dataset_count = 3;
    int64 raw_bytes = 4;
    int64 zip_bytes = 5;
    int32 job_count = 6;
}

message Expert {
    string expert_id = 1;
    string expert_name = 2;
}

message DatasetData {
    string dataset_id = 1;
    string name = 2;
    string project_id = 3;

    int64 record_count = 4;
    bool record_count_set = 5;

    int64 file_count = 6;
    bool file_count_set = 7;

    int64 raw_bytes = 8;
    bool raw_bytes_set = 9;

    int64 zip_bytes = 10;
    bool zip_bytes_set = 11;



    bytes sample_body = 12;
    SAMPLE_KIND sample_kind = 13;
    bool sample_set = 14;

    int64 update_timestamp = 15;
    bool update_timestamp_set = 16;

    string data_format = 17;
    bool data_format_set = 18;
    string description = 19;
    bool description_set = 20;


    repeated string upstream_jobs = 21;
    repeated string downstream_jobs = 22;

    repeated Expert experts = 23;
}




message Job{
    string job_id = 1;
    string job_name = 2;
    string project_id = 3;

    repeated string inputs = 4;
    repeated string outputs = 5;

    repeated Expert experts = 6;
}





message TenantStats{
    int32 project_count = 1;
    int32 dataset_count = 2;
    int32 job_count = 3;
    int32 report_count = 4;
    int32 expert_count = 5;
    int32 model_count = 6;
}


message AssetCache{
    string digest = 1;
    bytes body = 2;
}


message ChunkData {
    string hash = 1;
    string name = 2;
    int64 raw_size = 3;
    int64 zip_size = 4;
}


enum LinkType {
    UNIVERSAL=0;
    UPLOAD=1;
}

message SourceLink {
    LinkType kind = 1;
    string source_name = 2;
    string user_name = 3;

}

message VersionData {
    int32 version = 1;
    repeated ChunkData chunks = 2;
    repeated SourceLink sources = 3;
    int64 raw_size = 4;
    int64 zip_size = 5;
    string dataset_id = 6;

    DatasetDescriptor descriptor = 7;

}



message FieldSchema {
    string name = 1;
    string type = 2;
}

message DatasetDescriptor {
    string type = 1;
    int64 record_count = 2;
    repeated FieldSchema schema = 3;
    string sample_string = 4;
}